indexing-service/src/main/java/io/druid/indexing/overlord/TaskRunnerListener.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.indexing.overlord;
+
+import io.druid.indexing.common.TaskLocation;
+
+import java.util.concurrent.Executor;
+
+/**
+ * Listener to be registered with {@link TaskRunner#registerListener(TaskRunnerListener, Executor)}.
+ */
+public interface TaskRunnerListener
```

Comment:
it might have new notification types added at some point (perhaps for status if we ever have a use case where that is more useful than using the futures from 'run'). but there are currently no use cases I am aware of other than location.